src/applications/travel-pay/tests/containers/TravelPayStatusApp.unit.spec.jsx

```diff
@@ -60,17 +76,126 @@ describe('App', () => {
   });
 
   it('should redirect if feature flag is off', async () => {
-    renderWithStoreAndRouter(<App />, initialStateFeatureFlag(false, false));
+    renderWithStoreAndRouter(<App />, {
+      initialState: getData({
+        areFeatureTogglesLoading: false,
+        hasFeatureFlag: false,
+      }),
+      path: `/`,
+      reducers: reducer,
+    });
     await waitFor(() => {
       expect(window.location.replace.called).to.be.true;
     });
   });
 
-  it('should render loading state if feature flag is loading', () => {
-    const screenFeatureToggle = renderWithStoreAndRouter(
-      <App />,
-      initialStateFeatureFlag(),
-    );
-    expect(screenFeatureToggle.getByTestId('travel-pay-loading-indicator'));
+  it('should render loading state if feature flag is loading', async () => {
+    const screenFeatureToggle = renderWithStoreAndRouter(<App />, {
+      initialState: getData(),
+      path: `/`,
+      reducers: reducer,
+    });
+    expect(
+      await screenFeatureToggle.getByTestId('travel-pay-loading-indicator'),
+    ).to.exist;
+  });
+
+  it('renders a login prompt for an unauthenticated user', async () => {
+    const screen = renderWithStoreAndRouter(<App />, {
+      initialState: getData({
+        areFeatureTogglesLoading: false,
+        hasFeatureFlag: true,
+        isLoggedIn: false,
+      }),
+      path: `/`,
+      reducers: reducer,
+    });
+    expect(await screen.findByText('Log in to view your travel claims')).to
+      .exist;
+  });
+
+  it('shows the login modal when clicking the login prompt', async () => {
+    const { container } = renderWithStoreAndRouter(<App />, {
+      initialState: getData({
+        areFeatureTogglesLoading: false,
+        hasFeatureFlag: true,
+        isLoggedIn: true,
+      }),
+      path: `/`,
+      reducers: reducer,
+    });
+
+    expect($('va-loading-indicator', container)).to.exist;
+  });
+
+  it('handles a failed fetch of claims', async () => {
+    global.fetch.restore();
+    mockApiRequest({ error: 'oh no' }, false);
+
+    const screen = renderWithStoreAndRouter(<App />, {
+      initialState: getData({
+        areFeatureTogglesLoading: false,
+        hasFeatureFlag: true,
+        isLoggedIn: true,
+      }),
+      path: `/`,
+      reducers: reducer,
+    });
+
+    await waitFor(() => {
+      expect(screen.queryAllByTestId('travel-claim-details').length).to.eq(0);
+    });
+  });
+
+  it('shows the login modal when clicking the login prompt', async () => {
+    const { container } = renderWithStoreAndRouter(<App />, {
+      initialState: getData({
+        areFeatureTogglesLoading: false,
+        hasFeatureFlag: true,
+        isLoggedIn: false,
+      }),
+      path: `/`,
+      reducers: reducer,
+    });
+
+    fireEvent.click($('va-button', container));
+    // TODO: make this check for the modal itself
+    expect($('va-button', container)).to.exist;
```

Comment:
Since this test file doesn't actually include the sign-in modal import, this is a rather brittle way of testing it. I'd like to figure out the best way to improve this.